powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/repository/OperationRepository.java

```diff
@@ -47,13 +47,26 @@ public interface OperationRepository extends CrudRepository<OperationEntity, Str
     @Query("SELECT o FROM OperationEntity o WHERE o.id = :operationId")
     Optional<OperationEntity> findOperation(String operationId);
 
-    @Query("SELECT DISTINCT o FROM OperationEntity o INNER JOIN o.applications a WHERE o.userId = :userId AND a.id in :applicationIds ORDER BY o.timestampCreated DESC")
-    Stream<OperationEntity> findAllOperationsForUser(String userId, List<String> applicationIds, final Pageable pageable);
+    @Query("""
+       SELECT DISTINCT o FROM OperationEntity o INNER JOIN o.applications a
+       WHERE o.userId = :userId
+       AND a.id in :applicationIds
+       AND (:activationId IS NULL OR o.activationId = :activationId OR o.activationId IS NULL)
+       AND (:activationFlags IS NULL OR o.activationFlag IN :activationFlags OR o.activationFlag IS NULL)
```

Comment:
Isn't the order important?
```suggestion
       AND (:activationFlags IS NULL OR o.activationFlag IS NULL OR o.activationFlag IN :activationFlags)
```

Reply:
IMHO I do not think it is, it is OR statement. I do agree it might be better to switch it for better readability.